### PR TITLE
Feat in app code editor

### PR DESCRIPTION
**Pull request by davidsoderberg**
### What change does this PR introduce?

Adding code editor, variable list and preview for in app

### Other information (Screenshots)
Screenshot 2023-02-01 at 09 09 25
Screenshot 2023-02-01 at 09 09 32
Screenshot 2023-02-01 at 09 09 50

### CODE REVIEW

File: apps/api/src/app/events/usecases/send-message/send-message-in-app.usecase.ts [hunk 1]
```diff
@@ -68,26 +70,41 @@ export class SendMessageInApp extends SendMessageBase {
     const { actor } = command.step.template;
 
     if (actor && actor.type !== ActorTypeEnum.NONE) {
-      actor.data = await this.processAvatar(actor, command, notification);
+      actor.data = await this.processAvatar(actor, command);
     }
 
+    const organization = await this.organizationRepository.findById(command.organizationId);
```
Comment (p-fernandez) [inline apps/api/src/app/events/usecases/send-message/send-message-in-app.usecase.ts:76]: What do you think about using the `GetOrganization` use case from the Organization module and bring it as a dependency to the Events module? So we can start to isolate the DAL dependencies in their corresponding module (Organization in this case)?

File: apps/api/src/app/events/usecases/send-message/send-message-in-app.usecase.ts [hunk 2]
```diff
@@ -247,17 +265,17 @@ export class SendMessageInApp extends SendMessageBase {
...
           ...payload,
         },
       })
     );
   }
 
-  private async processAvatar(
-    actor: IActor,
-    command: SendMessageCommand,
-    notification: NotificationEntity
```
Comment (p-fernandez) [inline apps/api/src/app/events/usecases/send-message/send-message-in-app.usecase.ts:259]: Good catch.

File: apps/api/src/app/events/usecases/send-message/send-message-in-app.usecase.ts [hunk 3]
```diff
@@ -247,17 +265,17 @@ export class SendMessageInApp extends SendMessageBase {
             events: command.events,
             total_count: command.events?.length,
           },
+          branding: {
+            logo: organization?.branding?.logo,
+            color: organization?.branding?.color || '#f47373',
```
Comment (p-fernandez) [inline apps/api/src/app/events/usecases/send-message/send-message-in-app.usecase.ts:270]: When creating an organization we don't brand it by default? Logo for sure not, but I thought the color yes.

File: apps/web/src/components/templates/VariableManagerModal.tsx [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+export const VariableManagerModal = ({
+  open,
+  setOpen,
+  index,
+  variablesArray,
+}: {
+  open: boolean;
+  setOpen: (open: boolean) => void;
+  index: number;
+  variablesArray: any;
```
Comment (p-fernandez) [inline apps/web/src/components/templates/VariableManagerModal.tsx:14]: Is this maybe typed with `ITemplateVariable[]`?

File: apps/web/src/components/templates/in-app-editor/AvatarFeedFields.tsx [hunk 5]
```diff
@@ -0,0 +1,136 @@
...
+
+import { Checkbox, colors, Input } from '../../../design-system';
+import { useEnvController } from '../../../store/useEnvController';
+import { createFeed, getFeeds } from '../../../api/feeds';
+import { QueryKeys } from '../../../api/query.keys';
+import { PlusGradient } from '../../../design-system/icons';
+import { FeedItems } from './FeedItems';
+import { EnableAvatarSwitch } from './EnableAvatarSwitch';
+
+export const AvatarFeedFields = ({ index, control }) => {
```
Comment (p-fernandez) [inline apps/web/src/components/templates/in-app-editor/AvatarFeedFields.tsx:17]: I struggle to understand what's the relation with Avatars for this component besides the use of `EnableAvatarSwitch` inside. What is this component goal?

File: apps/web/src/components/templates/in-app-editor/InAppEditorBlock.tsx [hunk 6]
```diff
@@ -28,42 +33,51 @@ export function InAppEditorBlock({
...
-              readonly={readonly}
-            />
+          <InAppWidgetPreview
+            {...fieldRefs}
+            preview={preview}
+            readonly={readonly}
+            enableAvatar={!!enableAvatar}
+            index={index}
+          >
+            <>
```
Comment (p-fernandez) [inline apps/web/src/components/templates/in-app-editor/InAppEditorBlock.tsx:43]: Reading the conditions of the `When` components I think the React fragment component (<>) is not needed.

File: apps/web/src/components/widget/InAppWidgetPreview.tsx [hunk 7]
```diff
@@ -42,6 +44,83 @@ export function InAppWidgetPreview({
...
+            content: '""',
+            position: 'absolute',
+            left: '0',
+            top: '0',
+            right: '0',
+            bottom: '0',
+            width: '5px',
+            borderRadius: ' 7px 0 0 7px',
+            background: colors.vertical,
+          },
```
Comment (p-fernandez) [inline apps/web/src/components/widget/InAppWidgetPreview.tsx:75]: Is it worth to use Mantine's `createStyles` for this?

File: apps/api/src/app/events/usecases/send-message/send-message-in-app.usecase.ts (hunk 3 above)
Comment (davidsoderberg) [reply apps/api/src/app/events/usecases/send-message/send-message-in-app.usecase.ts:270]: No we do this trick instead...

File: apps/web/src/components/widget/InAppWidgetPreview.tsx (hunk 7 above)
Comment (davidsoderberg) [reply apps/web/src/components/widget/InAppWidgetPreview.tsx:75]: yeah this just refactoring :)

File: apps/web/src/components/templates/in-app-editor/InAppEditorBlock.tsx (hunk 6 above)
Comment (davidsoderberg) [reply apps/web/src/components/templates/in-app-editor/InAppEditorBlock.tsx:43]: the `InAppWidgetPreview` component ask for only one element

File: apps/web/src/components/templates/in-app-editor/AvatarFeedFields.tsx (hunk 5 above)
Comment (davidsoderberg) [reply apps/web/src/components/templates/in-app-editor/AvatarFeedFields.tsx:17]: It is a grouping for avatar fields and feed fields

File: apps/api/src/app/events/usecases/send-message/send-message-in-app.usecase.ts [hunk 8]
```diff
@@ -46,7 +47,7 @@ export class SendMessageInApp extends SendMessageBase {
     protected createExecutionDetails: CreateExecutionDetails,
     protected subscriberRepository: SubscriberRepository,
     private compileTemplate: CompileTemplate,
-    private organizationRepository: OrganizationRepository
+    private getOrganization: GetOrganization
```
Comment (p-fernandez) [inline apps/api/src/app/events/usecases/send-message/send-message-in-app.usecase.ts:50]: 👏🏻

**Review by p-fernandez — APPROVED** [review]
🌟

**Pull request merged**